### PR TITLE
Fix entity column reads

### PR DESCRIPTION
**Pull request by dmdashenkov**
This PR fixes two bugs found in the entity column querying mechanism:
 - No additional in-memory predicate is applied to the filtered by columns query results now.
 - If both IDs and columns are specified, an in-memory predicate is applied to the by-IDs query results.

Also, the version of Spine has been increased up to `0.10.11-SNAPSHOT`.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: datastore/build.gradle [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ext {
     artifactId = 'gae-java'
+    datastoreVersion = '1.12.0'
```
Comment (armiol) [inline datastore/build.gradle]: Can we please have this version in `ext.gradle`.

File: datastore/src/main/java/io/spine/server/storage/datastore/DsRecordStorage.java [hunk 2]
```diff
@@ -245,7 +246,10 @@ public EntityRecord apply(@Nullable Entity input) {
         );
         Iterator<EntityRecord> result = emptyIterator();
         for (StructuredQuery<Entity> query : queries) {
-            final Iterator<EntityRecord> records = queryAll(typeUrl, query, fieldMask);
+            final Iterator<EntityRecord> records = queryAll(typeUrl,
```
Comment (armiol) [inline datastore/src/main/java/io/spine/server/storage/datastore/DsRecordStorage.java]: The method, in which body this comment resides, is already way too complex and needs to be refactored into something much more readable. I suggest to have a few well-commented methods instead.

**Review by armiol — CHANGES_REQUESTED** [review]
@dmdashenkov LGTM in general. Please address the comments and submit again.

**Comment by dmdashenkov** [comment]
@armiol, PTAL again.

File: datastore/src/main/java/io/spine/server/storage/datastore/DsRecordStorage.java [hunk 3]
```diff
@@ -212,40 +213,92 @@ public EntityRecord apply(@Nullable Entity input) {
...
+            inMemPredicate = alwaysTrue();
+        }
+        final Iterator<EntityRecord> records = lookup(acceptableIds, fieldMask, inMemPredicate);
+        return records;
+    }
+
+    /**
+     * Performs a query by entity columns.
+     *
+     * <p>The query is performed on Datastore. A single call to this method may turn into a few API
```
Comment (armiol) [inline datastore/src/main/java/io/spine/server/storage/datastore/DsRecordStorage.java]: `...into several API calls.` Let's discuss the difference in person.

File: datastore/src/main/java/io/spine/server/storage/datastore/DsRecordStorage.java [hunk 4]
```diff
@@ -212,40 +213,92 @@ public EntityRecord apply(@Nullable Entity input) {
...
-        );
+        final FilterToQuery function = new FilterToQuery(datastoreQuery);
+        final Collection<StructuredQuery<Entity>> queries = transform(filters, function);
+        final Iterator<EntityRecord> result = queryAndMerge(queries, fieldMask);
+        return result;
+    }
+
+    /**
+     * Performs the given Datastore {@linkplain StructuredQuery queries} and combines results into
+     * a single lazy iterator.
```
Comment (armiol) [inline datastore/src/main/java/io/spine/server/storage/datastore/DsRecordStorage.java:288]: Please explain the lazy nature in the following paragraphs.

**Review by armiol — COMMENTED** [review]
@dmdashenkov please see my comments.

File: datastore/src/main/java/io/spine/server/storage/datastore/DsRecordStorage.java (hunk 4 above)
Comment (dmdashenkov) [reply datastore/src/main/java/io/spine/server/storage/datastore/DsRecordStorage.java:288]: Done.

**Comment by dmdashenkov** [comment]
@armiol, PTAL again.

**Review by armiol — APPROVED** [review]
@dmdashenkov LGTM.

**Pull request merged**